Importing required libraries 

In [89]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np 

Loading the dataset

In [90]:
job_data = pd.read_csv('Data/Dataset.csv')

Displaying basic information about the dataset

In [91]:
job_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9355 entries, 0 to 9354
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           9355 non-null   int64 
 1   job_title           9355 non-null   object
 2   job_category        9355 non-null   object
 3   salary_currency     9355 non-null   object
 4   salary              9355 non-null   int64 
 5   salary_in_usd       9355 non-null   int64 
 6   employee_residence  9355 non-null   object
 7   experience_level    9355 non-null   object
 8   employment_type     9355 non-null   object
 9   work_setting        9355 non-null   object
 10  company_location    9355 non-null   object
 11  company_size        9355 non-null   object
dtypes: int64(3), object(9)
memory usage: 877.2+ KB


In [92]:
job_data.head(10)

work_year                    job_title                    job_category  \
0       2023         Data DevOps Engineer                Data Engineering   
1       2023               Data Architect  Data Architecture and Modeling   
2       2023               Data Architect  Data Architecture and Modeling   
3       2023               Data Scientist       Data Science and Research   
4       2023               Data Scientist       Data Science and Research   
5       2023               Data Scientist       Data Science and Research   
6       2023               Data Scientist       Data Science and Research   
7       2023  Machine Learning Researcher         Machine Learning and AI   
8       2023  Machine Learning Researcher         Machine Learning and AI   
9       2023                Data Engineer                Data Engineering   

  salary_currency  salary  salary_in_usd employee_residence experience_level  \
0             EUR   88000          95012            Germany        Mid-level   
1             USD  186000         186000      United States           Senior   
2             USD   81800          81800      United States           Senior   
3             USD  212000         212000      United States           Senior   
4             USD   93300          93300      United States           Senior   
5             USD  130000         130000      United States           Senior   
6             USD  100000         100000      United States           Senior   
7             USD  224400         224400      United States        Mid-level   
8             USD  138700         138700      United States        Mid-level   
9             USD  210000         210000      United States        Executive   

  employment_type work_setting company_location company_size  
0       Full-time       Hybrid          Germany            L  
1       Full-time    In-person    United States            M  
2       Full-time    In-person    United States            M  
3       Full-time    In-person    United States            M  
4       Full-time    In-person    United States            M  
5       Full-time       Remote    United States            M  
6       Full-time       Remote    United States            M  
7       Full-time    In-person    United States            M  
8       Full-time    In-person    United States            M  
9       Full-time       Remote    United States            M

Data cleaning - Missing Values

In [93]:
job_data.isnull().sum()

work_year             0
job_title             0
job_category          0
salary_currency       0
salary                0
salary_in_usd         0
employee_residence    0
experience_level      0
employment_type       0
work_setting          0
company_location      0
company_size          0
dtype: int64

Data cleaning - Checking Duplicates and drop duplicate rows

In [94]:
print("Number of duplicate rows:", job_data.duplicated().sum())
job_data.drop_duplicates(inplace=True)

Number of duplicate rows: 4014


In [95]:
job_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5341 entries, 0 to 9354
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           5341 non-null   int64 
 1   job_title           5341 non-null   object
 2   job_category        5341 non-null   object
 3   salary_currency     5341 non-null   object
 4   salary              5341 non-null   int64 
 5   salary_in_usd       5341 non-null   int64 
 6   employee_residence  5341 non-null   object
 7   experience_level    5341 non-null   object
 8   employment_type     5341 non-null   object
 9   work_setting        5341 non-null   object
 10  company_location    5341 non-null   object
 11  company_size        5341 non-null   object
dtypes: int64(3), object(9)
memory usage: 542.4+ KB


Before : 9355 entries, 0 to 9354
Data columns (total 12 columns)

After: 5341 entries, 0 to 9354
Data columns (total 12 columns)

Number of duplicate rows Removed: 4014

Descriptive statistics

In [96]:
job_data.describe()

work_year         salary  salary_in_usd
count  5341.000000    5341.000000    5341.000000
mean   2022.682082  145814.937839  146258.409099
std       0.608026   67025.469452   66594.117529
min    2020.000000   14000.000000   15000.000000
25%    2022.000000   97300.000000   98506.000000
50%    2023.000000  140000.000000  140000.000000
75%    2023.000000  186200.000000  186000.000000
max    2023.000000  450000.000000  450000.000000

Save cleaned dataset 

In [97]:
job_data.to_csv('Data/Cleaned_Dataset.csv', index=False)

Loading the cleaned dataset

In [98]:
cleaned_data = pd.read_csv('Data/Cleaned_Dataset.csv')

In [99]:
cleaned_data.head()

work_year             job_title                    job_category  \
0       2023  Data DevOps Engineer                Data Engineering   
1       2023        Data Architect  Data Architecture and Modeling   
2       2023        Data Architect  Data Architecture and Modeling   
3       2023        Data Scientist       Data Science and Research   
4       2023        Data Scientist       Data Science and Research   

  salary_currency  salary  salary_in_usd employee_residence experience_level  \
0             EUR   88000          95012            Germany        Mid-level   
1             USD  186000         186000      United States           Senior   
2             USD   81800          81800      United States           Senior   
3             USD  212000         212000      United States           Senior   
4             USD   93300          93300      United States           Senior   

  employment_type work_setting company_location company_size  
0       Full-time       Hybrid          Germany            L  
1       Full-time    In-person    United States            M  
2       Full-time    In-person    United States            M  
3       Full-time    In-person    United States            M  
4       Full-time    In-person    United States            M